### ISSUE TITLE
Option to display "Miss" floating text, enabled by default for beginners

### ISSUE THREAD

**Issue by Dalas121** [wesnoth/wesnoth#9461]
### Describe the desired feature

My TSG revision and TDG campaign include floating "Miss" text after a missed strike, to help clarify hits/misses for beginners - this was a common point of confusion I noticed in my (limited) new player playtesting.

![Missed Attacks](https://github.com/user-attachments/assets/63b5b912-6f81-4f3e-b4f3-dc893d573acc)

It was suggested in the discord that this be made an option in the core game.  If possible I'd prefer that this option default **off** for all existing players but default **on** for new installs, with a checkbox in Preferences->Display to toggle it.

Is this something that's not too much trouble to add?

**Comment by CelticMinstrel** [comment]
Related: I wonder if it's worth revisiting the discussion of whether it's okay to grant add-ons access to your preferences.

Why is it related, you ask?

Because if that were available, this could theoretically be done as an add-on that adds an `[advanced_preference]` and reads the value in the add-on.

(I'm not sure if advanced preferences will actually be loaded from add-ons though. That would need to be checked too if we want this.)

Of course, if done that way the checkbox would be in advanced preferences, not in Display, which might be an unacceptable loss…

**Comment by Dalas121** [comment]
Anyone interested or able to work on this? I don't know how challenging this is to implement - I know it's a lot easier to request changes than to make them - but I do think it would be much better to have this as a proper preference rather than something hardcoded into a couple beginner campaigns that never shows up again.

The addons-access-to-preferences solution could potentially also work, and I'm sure that would be beneficial to a lot of add-on creators as well.

**Comment by Gothyoba** [comment]
Temporarily, since I don’t think anyone is actively working on this, could the text be added to all novice camoaigns and TDG so there is a clear guideline to where it should be added, rather than just be added to a few campaigns in a seemingly somewhat arbitrary way.

**Comment by Vultraz** [comment]
Implemented. The new option is in advanced preferences.

**Comment by Dalas121** [comment]
Wow, thank you very much! I'll remove the event-based text from TSG.

**Comment by ProditorMagnus** [comment]
It is not enabled by default in your campaign anymore though.

**Comment by Dalas121** [comment]
Ah, I see. I'm guessing it would be very challenging to make this option behave as suggested in the original issue: default off for existing players but default on for new installs?

**Comment by Vultraz** [comment]
Yeah, it either has to be one or the other. I could enable it globally and existing players just turn it off 😬

**Comment by Dalas121** [comment]
Whatever you think is best, then. I don't think "Miss" text is much use for new players when it's off by default, but I also don't want to frustrate veteran players who liked things the old way - I don't think there's any great answer.
